apps/agreement/contracts/Agreement.sol

```diff
@@ -712,6 +745,46 @@ contract Agreement is IAgreement, AragonApp {
 
     // Internal fns
 
+    /**
+    * @dev Pay transactions fees required for new actions
+    * @param _settingId Identification number of the setting being queried
+    * @param _disputable Address of the Disputable app, used to determine fees
+    * @param _staking Staking pool for the ERC20 token to transfer fees from
+    * @param _submitter Address of the user that has submitted the action
+    * @param _actionId Identification number of the action to be paid for
+    */
+    function _payAppFees(uint256 _settingId, IDisputable _disputable, Staking _staking, address _submitter, uint256 _actionId) internal {
+        // Get fees
+        Setting storage setting = _getSetting(_settingId);
+        IAragonAppFeesCashier aragonAppFeesCashier = setting.aragonAppFeesCashier;
+        if (address(aragonAppFeesCashier) == address(0)) {
```

Comment:
Not a big fan of casting address zero to a contract/interface, because it’s a “fake” casting and could potentially lead to issues, as it won’t actually have any of the assumed methods, but for this comparison it doesn’t seem important.